### PR TITLE
fixes exception handling and adds basic logging to http client

### PR DESCRIPTION
**Pull request by pawelgocek**
Exceptions that didn't have JSON body were not thrown properly by HTTP client. This PR fixes error handling and adds basic logging to HTTP client.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -91,4 +92,11 @@ bintray {
         labels = ['base crm', 'api', 'http', 'v2', 'client', 'HTTP client']
         publicDownloadNumbers = true
     }
+}
+
+test {
+    if (!System.getenv('BASECRM_ACCESS_TOKEN')) {
```
Comment (michal-mally) [inline build.gradle:98]: We shall not by default run tests that require external dependencies (e.g. Base CRM application running and test account on it). By default only unit tests shall be running which should not require Internet connection or any other external dependency.

**Comment by michal-mally** [comment]
@pawelgocek see my comments

File: build.gradle (hunk 1 above)
Comment (pawelgocek) [inline build.gradle:98]: Agree but this is out of scope for this enhancement.
